### DISCUSSION TITLE
GAS m3 readings current and previous are not aligning.

### DISCUSSION THREAD

**Discussion by andyblac** [BottlecapDave/HomeAssistant-OctopusEnergy#702] [General]
as tile the 2 are not the same, how are these figures got / calculated ?

Current Consumption for 10th
<img width="243" alt="Screenshot 2024-01-11 at 18 55 24" src="https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/assets/77936/bb8af121-2ab0-4706-9414-63375eef463d">

previous Consumption for the 10th ( data date 11th)
<img width="264" alt="Screenshot 2024-01-11 at 18 55 37" src="https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/assets/77936/060bff39-04b5-41da-a08c-6983288d6123">

**Comment by BottlecapDave** [comment]
These can vary for a number of reasons.

1. The [current consumption](https://bottlecapdave.github.io/HomeAssistant-OctopusEnergy/entities/gas/#current-accumulative-consumption-m3) is calculated/estimated due to the native data from the API being in kWh. This is done using an [forumla](https://www.theenergyshop.com/guides/how-to-convert-gas-units-to-kwh) where the calorific value can differ per time of year. There is nothing in the API to get this automatically, so needs to be configured [manually](https://bottlecapdave.github.io/HomeAssistant-OctopusEnergy/setup/account/#calorific-value).
2. As per the [FAQ](https://bottlecapdave.github.io/HomeAssistant-OctopusEnergy/faq/#data-in-my-home-assistant-energy-dashboard-reported-by-octopus-home-mini-differs-to-octopus-energy-dashboard-why-is-this) the data can differ between the home mini and what's reported by your meter (despite them theoretically being from the same source). Depending on what your meter reports in natively, the data from previous consumptions sensor m3 might be straight from your meter or calculated using the above way. This is indicated in the sensor by the `is_estimated` flag

**Comment by andyblac** [comment]
hmm, do you have the formula so I find what cal value octpus is using as I have enter the last one from my bill, but it is obviously wrong.

**Reply by BottlecapDave** [reply]
Formula is linked above for kWh to m3. Code can be found [here](https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/blob/develop/custom_components/octopus_energy/gas/__init__.py#L12). Octopus Energy formula for m3 to kWh can be found at https://octopus.energy/blog/statement-questions/#unitrates and can be found on the bill.

As mentioned, the most likely scenario is a discrepency in what the Home Mini reported and what your meter reported via the DCC.

**Comment by andyblac** [comment]
yeah thtats what I using, but if I enter the m3 the kWh does not match what is used in their app on website.

ie my bill says 40.3, but to convert m3 to kWh to mach what their app / website shows I have to use 39.50, I used this your integration (also tried 40.3) nether match kWh to m3 conversion. looks like some thing is not right., as I can't get both to link up.

**Reply by BottlecapDave** [reply]
hmm. I think maybe there's something wrong with the calculation (that OE have highlighted or the final value) as if I put the calculation I have on my last bill into Google I got a slightly different value (lower) to what's on my bill

What do you get if you put the calculation on your bill into a calculator or Google?

**Reply by BottlecapDave** [reply]
The only thing I can think of is that the calorific value that is used is more specific (as the bill states its rounded to 1 decimal place) and that is causing the offset. For instance mine is 38.9 on the bill, but I don't get close enough to the value on my bill unless I put in 38.906

**Comment by andyblac** [comment]
here's my spreadsheet I have 

this one is cal value of `39.5` so both m3 and kWh match that of the app.
<img width="389" alt="Screenshot 2024-01-11 at 19 49 32" src="https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/assets/77936/bbf35a79-43ab-4bf1-811e-2c2cd2d78798">

here is my last total m3 from last bill using same 39.5
<img width="221" alt="Screenshot 2024-01-11 at 19 52 23" src="https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/assets/77936/eb17f4d2-f54d-41cc-b072-470645c82a02">

here what my says
<img width="789" alt="Screenshot 2024-01-11 at 19 53 00" src="https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/assets/77936/cc23b6d6-dd3e-4a09-9c35-1093c02f9ef2">

and if I use 40.3 in spreadsheet
<img width="208" alt="Screenshot 2024-01-11 at 19 54 13" src="https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/assets/77936/1fd6618d-f337-46e5-8241-49c86139ae7a">

here is the app for this week
![IMG_2327](https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/assets/77936/90e87fc0-34b7-4f4d-add1-6d9955c5a8b0)

**Comment by andyblac** [comment]
so must be the app / website are using a standard value for all people, maybe it should be treated and a ROUGH estimate and not what your ACUTAL cost / usage is ???

**Reply by BottlecapDave** [reply]
My calculation on my last bill is `184.3 × 1.02264 × 38.9 ÷ 3.6 = 2036.9`, however if I put that calculation into Google or local calculator I get `2036.55063` as the answer.

If the app/website is using web technologies, there is also floating point precision that will come into it, which will produce inaccuracies in calculations. Something that python may also suffer a little bit from.

As mentioned though, my guess is the calorific value is probably more accurate that what is stated on the bill (which is why when you use the specified value you're .119 out.

And in regards to the differences between current and previous consumption, as mentioned the home mini does sometimes report different amounts but ultimately what your meter specifies from the DCC is what's on your bill. This is what comes back in the previous consumption sensors.
